chocs/console/dto_generator/processor.py

```diff
@@ -0,0 +1,178 @@
+        for property_name, property_schema in schema["properties"].items():
+            dto_property = AstPropertyNode(property_name)
+            dto_class.properties[property_name] = dto_property
+
+            if property_name in schema.get("required", []):
+                dto_property.optional = False
+
+            try:
+                dto_property.type = self._process_type_node(property_schema)
+            except RuntimeError as e:
+                raise RuntimeError(f"Cannot process property {property_name}") from e
+
+    def _process_type_node(self, schema: Union[JsonReference, Dict[str, Any]]) -> AstTypeNode:
+        if isinstance(schema, JsonReference):
+            return AstReferenceTypeNode(self._process_class_node(schema.id, schema))
+
+        if "type" not in schema:
+            return AstTypeNode("str")
+
+        if schema["type"] == "object":
+            return AstTypeNode("typing.Dict[str, typing.Any]")
+
+        if schema["type"] == "integer":
+            return AstTypeNode("int")
+
+        if schema["type"] in ("float", "number"):
+            return AstTypeNode("float")
+
+        if schema["type"] == "boolean":
+            return AstTypeNode("bool")
+
+        if schema["type"] == "string":
+            if "format" not in schema:
+                return AstTypeNode("str")
+
+            if schema["format"] == "date":
+                return AstTypeNode("datetime.date")
+
+            if schema["format"] == "date-time":
+                return AstTypeNode("datetime.datetime")
+
+            if schema["format"] == "time":
+                return AstTypeNode("datetime.time")
+
+            if schema["format"] == "ip-address-v4":
+                return AstTypeNode("ipaddress.IPv4Address")
+
+            if schema["format"] == "ip-address-v6":
+                return AstTypeNode("ipaddress.IPv6Address")
+
+            if schema["format"] == "time-duration":
+                return AstTypeNode("datetime.timedelta")
+
+            if schema["format"] == "decimal":
+                return AstTypeNode("decimal.Decimal")
+
+            if schema["format"] == "byte":
+                return AstTypeNode("bytes")
+
+            return AstTypeNode("str")
+
+        if schema["type"] == "array":
+            property_type = AstListTypeNode("list")
+
+            if "uniqueItems" in schema:
+                property_type.unique = schema["uniqueItems"]
+
+            if "items" not in schema:
+                property_type.items.append(AstTypeNode("typing.Any"))
+                return property_type
+
+            if isinstance(schema["items"], list):
+                property_type.items = [self._process_type_node(item_schema) for item_schema in schema["items"]]
+            else:
+                property_type.items = [self._process_type_node(schema["items"])]
+            return property_type
+
+        raise RuntimeError(f"Unknown property type {schema['type']}")
```

Comment:
When I see this, I can't help but think about how to go about translating it into a table. But that is an optimization to make in the future.